File: lakeSide-hotel-demo-client-main/src/model/Status.js
class Status {
    constructor(id = null, startDate = null, endDate = null, status = null, 
        description = null, room = null) {
        this.id = id;
        this.startDate = startDate; // Kiểu Date
        this.endDate = endDate; // Kiểu Date
        this.status = status; // Chuỗi (String)
        this.description = description; // Chuỗi (String)
        this.room = room; // Đối tượng của class RoomModel
    }
}

export default Status;

File: lakeSide-hotel-demo-server-master/src/main/java/com/dailycodework/lakesidehotel/model/Status.java
package com.dailycodework.lakesidehotel.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

/**
 * @author Simpson Alfred
 */
@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Status {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private  Long id;

    @Column(name = "start_date")
    private LocalDate startDate;

    @Column(name = "end_date")
    private LocalDate endDate;

    @Column(name = "status")
    private String status;

    @Column(name = "description", columnDefinition = "TEXT")
    private String description;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "room_id")
    private Room room;


    public Status(LocalDate startDate, LocalDate endDate, String status, String description) {
        this.startDate = startDate;
        this.endDate = endDate;
        this.status = status;
        this.description = description;
    }
}

